keras_core/layers/preprocessing/category_encoding_test.py

```diff
@@ -55,7 +82,44 @@ def test_one_hot(self):
         self.assertAllClose(expected_output, output_data)
         self.assertEqual(expected_output_shape, output_data.shape)
 
+    def test_batched_one_hot(self):
+        input_data = np.array([[3, 2, 0, 1], [3, 2, 0, 1]])
+        expected_output = np.array(
+            [[
+                [0, 0, 0, 1],
+                [0, 0, 1, 0],
+                [1, 0, 0, 0],
+                [0, 1, 0, 0],
+            ],
+            [
+                [0, 0, 0, 1],
+                [0, 0, 1, 0],
+                [1, 0, 0, 0],
+                [0, 1, 0, 0],
+            ],
+            ]
+        )
+        num_tokens = 4
+        expected_output_shape = (2, num_tokens, num_tokens)
+
+        # Test call on layer directly.
+        layer = layers.CategoryEncoding(
+            num_tokens=num_tokens, output_mode="one_hot"
+        )
+        output_data = layer(input_data)
+        self.assertAllClose(expected_output, output_data)
+        self.assertEqual(expected_output_shape, output_data.shape)
+
     def test_tf_data_compatibility(self):
+        # This test case will fail if
+        # it ran while the backend is not TF
```

Comment:
It is supposed to work with all backends -- that's the entire point of the TFDataLayer. Presumably there's a bug.

Reply:
Yes, there was a bug, when the backend was switching to TF in `TFDataLayer.__call__` function, the backend did not switch in `CategoryEncoding` because I was setting the `output_function` to a specific function in the `__init__` and when the backend changed, the `output_function` remained with the same function and did not get updated, I solved that in the `self._encode`. I am glad that this bug happened because I understood a lot about the codebase.